Source/Schema.NET/ValuesConverter.cs

```diff
@@ -40,8 +34,147 @@ public class ValuesConverter : JsonConverter
             JsonReader reader,
             Type objectType,
             object existingValue,
-            JsonSerializer serializer) =>
-            throw new NotImplementedException();
+            JsonSerializer serializer)
+        {
+            var mainType = UnwrapNullable(objectType);
+
+            object argument = null;
+
+            var tokenType = reader.TokenType;
+            var value = SanitizeReaderValue(reader, tokenType);
+
+            JToken token = JToken.Load(reader);
+            if (mainType.GenericTypeArguments.Length == 1)
+            {
+                var type = UnwrapNullable(mainType.GenericTypeArguments[0]);
+                if (tokenType == JsonToken.StartArray)
+                {
+                    var listType = typeof(List<>).MakeGenericType(type);
+                    if (token.Any(t => !string.IsNullOrEmpty(GetTypeNameFromToken(t))))
```

Comment:
If the collection has a child with the @type attribute, we create a list with the base type, create each child with the correct type and add it. We need to resort to reflection, because the most common base class would be Thing and it clashes with the list type.